packages/common/test/grpc-operation.js

```diff
@@ -0,0 +1,337 @@
+      grpcOperation.on('complete', util.noop);
+
+      assert.strictEqual(startPollingCallCount, 1);
+    });
+
+    it('should close when no more message listeners are bound', function() {
+      grpcOperation.on('complete', util.noop);
+      grpcOperation.on('complete', util.noop);
+      assert.strictEqual(grpcOperation.hasActiveListeners, true);
+
+      grpcOperation.removeListener('complete', util.noop);
+      assert.strictEqual(grpcOperation.hasActiveListeners, true);
+
+      grpcOperation.removeListener('complete', util.noop);
+      assert.strictEqual(grpcOperation.hasActiveListeners, false);
+    });
+  });
+
+  describe('startPolling_', function() {
+    var listenForEvents_;
+
+    before(function() {
+      listenForEvents_ = GrpcOperation.prototype.listenForEvents_;
+    });
+
+    after(function() {
+      GrpcOperation.prototype.listenForEvents_ = listenForEvents_;
+    });
+
+    beforeEach(function() {
+      GrpcOperation.prototype.listenForEvents_ = util.noop;
+      grpcOperation.hasActiveListeners = true;
+    });
+
+    afterEach(function() {
+      grpcOperation.hasActiveListeners = false;
+    });
+
+    it('should not call getMetadata if no listeners', function(done) {
+      grpcOperation.hasActiveListeners = false;
+
+      grpcOperation.getMetadata = done; // if called, test will fail.
+
+      grpcOperation.startPolling_();
+      done();
+    });
+
+    it('should call getMetadata if listeners are registered', function(done) {
+      grpcOperation.hasActiveListeners = true;
+
+      grpcOperation.getMetadata = function() {
+        done();
+      };
+
+      grpcOperation.startPolling_();
+    });
+
+    describe('API error', function() {
+      var error = new Error('Error.');
+
+      beforeEach(function() {
+        grpcOperation.getMetadata = function(callback) {
+          callback(error);
+        };
+      });
+
+      it('should emit the error', function(done) {
+        grpcOperation.on('error', function(err) {
+          assert.strictEqual(err, error);
+          done();
+        });
+
+        grpcOperation.startPolling_();
+      });
+    });
+
+    describe('operation failure', function() {
+      var formattedError = { status: 'a' };
+
+      var apiResponse = {
+        error: true,
+        result: 'b'
+      };
+
+      beforeEach(function() {
+        grpcOperation.getMetadata = function(callback) {
+          callback(null, apiResponse, apiResponse);
+        };
+      });
+
+      it('should emit the operation error', function(done) {
+        FakeGrpcService.decorateStatus_ = function(status) {
+          assert.strictEqual(status, apiResponse.result);
+          return formattedError;
+        };
+
+        grpcOperation.on('error', function(err) {
+          assert.strictEqual(err, formattedError);
+          done();
+        });
+
+        grpcOperation.startPolling_();
+      });
+    });
+
+    describe('operation pending', function() {
+      var apiResponse = { status: 'PENDING' };
```

Comment:
Since there is no `status`, we should just use `done: false` to indicate a running operation.